src/app/models/AttributeDecorators.ts

```diff
@@ -128,6 +128,12 @@ export function bawImage<Model>(
           defaultImage,
         ];
       } else if (imageUrls instanceof Array && imageUrls.length > 0) {
+
+        imageUrls = imageUrls.map((image) => ({
+            ...image,
+            ...defaultImage.url.includes(image.url) && {default: true},
+          }));
```

Comment:
I don't understand why this part is needed.

The default image, wherever it is used, has `default: true`, why then do you need to check the responses from the server. If the server sends any non-null image paths then they are by definition 'set' and not default

Reply:
Sorry, I should have added a comment saying that this PR is a WIP.

This code above is a very hacky work around that I created to test if other components would react to the default attribute correctly. I'm working towards the proper approach described above, but I'm still having troubles locating why I'm receiving extra default images (located in the `image_urls` property of the model).

After sending my patch setting `project: { image: null }`, the image_urls attribute on the model is still populated with `/images/project/project_span4.png`, `/images/project/project_span3.png`, etc.... I've been able to replicate this in the client & postman so I'm actively trying to narrow it down further.
e.g. Am I removing the images incorrectly, do I have to explicitly set `image_urls: null` or `image_urls: []`, etc...

Delete image PATCH request in (Postman)
![image](https://user-images.githubusercontent.com/33742269/210515643-542adb15-51f0-412f-b2bc-bb62a2ca83f8.png)

response from remove image PATCH request (Postman)
![image](https://user-images.githubusercontent.com/33742269/210515767-0b660bc4-7af5-4a2a-b4fb-de747d48f534.png)

GET request for the image model
![image](https://user-images.githubusercontent.com/33742269/210515061-76250d67-332f-40ee-834e-9321ef6e28e3.png)

Reply:
Ok, I found the root cause: https://github.com/QutEcoacoustics/baw-server/blob/b5a9f807d67a414610c665a499f39d50aa37d423/app/models/project.rb#L68

This will happen for [sites](https://github.com/QutEcoacoustics/baw-server/blob/b5a9f807d67a414610c665a499f39d50aa37d423/app/models/site.rb#L62), [projects](https://github.com/QutEcoacoustics/baw-server/blob/b5a9f807d67a414610c665a499f39d50aa37d423/app/models/project.rb#L68), and [users](https://github.com/QutEcoacoustics/baw-server/blob/b5a9f807d67a414610c665a499f39d50aa37d423/app/models/user.rb#L90), and is considered deprecated behavior. The hack you have here we might need to keep for the time being, but I'd probably suggest:

- matching urls with a regex: `\/images\/(.*)\/(\1)_span.*.png`
- documenting this fully
- filing an issue with the server to remove the default_url attribute

Reply:
Sounds like a good plan.

I agree using regex is 100% a more robust approach to this hack.

I'll be updating the PR, documenting, and filing an issue shortly.

Reply:
remember to document your hack including a link to the API issue you create

Reply:
I'll be accomplishing this through JsDoc comments, inline code comments, and `TODO: xyz` comments (along with linked API issues) as I don't think this warrants it's own wiki article.